lib/src/common/model/repos_statuses.dart

```diff
@@ -2,62 +2,83 @@ part of github.common;
 
 /// Model class for the combined status of a repository.
 class CombinedRepositoryStatus {
-  String state;
-  String sha;
-  int totalCount;
-  List<RepositoryStatus> statuses;
-  Repository repository;
+  final String state;
+  final String sha;
+  final int totalCount;
+  final List<RepositoryStatus> statuses;
+  final Repository repository;
 
-  CombinedRepositoryStatus();
+  CombinedRepositoryStatus._({
+    @required this.state,
+    @required this.sha,
+    @required this.totalCount,
+    @required this.statuses,
+    @required this.repository,
+  });
 
   static CombinedRepositoryStatus fromJSON(Map<String, dynamic> input) {
     if (input == null) return null;
 
-    return CombinedRepositoryStatus()
-      ..state = input["state"]
-      ..sha = input["sha"]
-      ..totalCount = input["total_count"]
-      ..statuses = (input["statuses"] as List<Map<String, dynamic>>)
+    return CombinedRepositoryStatus._(
+      state: input["state"],
+      sha: input["sha"],
+      totalCount: input["total_count"],
+      statuses: (input["statuses"] as List<Map<String, dynamic>>)
           .map((it) => RepositoryStatus.fromJSON(it))
-          .toList()
-      ..repository =
-          Repository.fromJSON(input["repository"] as Map<String, dynamic>);
+          .toList(),
+      repository:
+          Repository.fromJSON(input["repository"] as Map<String, dynamic>),
+    );
   }
 }
 
 /// Model class for the status of a repository at a particular reference.
 class RepositoryStatus {
-  DateTime createdAt;
-  DateTime updatedAt;
-  String state;
-  String targetUrl;
-  String description;
-  String context;
+  final DateTime createdAt;
+  final DateTime updatedAt;
+  final String state;
+  final String targetUrl;
+  final String description;
+  final String context;
+
+  RepositoryStatus._({
+    @required this.createdAt,
+    @required this.updatedAt,
+    @required this.state,
+    @required this.targetUrl,
+    @required this.description,
+    @required this.context,
+  });
 
-  static RepositoryStatus fromJSON(Map<String, dynamic> input) {
+  factory RepositoryStatus.fromJSON(Map<String, dynamic> input) {
     if (input == null) return null;
 
-    return RepositoryStatus()
-      ..createdAt = parseDateTime(input['created_at'])
-      ..updatedAt = parseDateTime(input['updated_at'])
-      ..state = input['state']
-      ..targetUrl = input['target_url']
-      ..description = input['description']
-      ..context = input['context'];
+    return RepositoryStatus._(
+      createdAt: parseDateTime(input['created_at']),
+      updatedAt: parseDateTime(input['updated_at']),
+      state: input['state'],
+      targetUrl: input['target_url'],
+      description: input['description'],
+      context: input['context'],
+    );
   }
 }
 
 /// Model class for a new repository status to be created.
+// TODO: this class is unused
```

Comment:
It looks like it has a usage here https://github.com/SpinlockLabs/github.dart/blob/master/lib/src/common/repos_service.dart#L1229
I think you can probably remove this TODO?